lib/haskell/natural4/src/LS/XPile/MathLang/GenericMathLang/TranslateL4.hs

```diff
@@ -570,16 +582,18 @@ baseExpify hl = case lhsLooksLikeLambda hl of
 
 mkLambda :: Reader Env :> es => [Var] -> BaseExp -> Eff es Exp
 mkLambda [] bexp = do
-  pos <- asks currSrcPos
-  pure $ typeMdata pos "Function" bexp
+  pos <- asks currentSrcPosition
+  pure $ typeMetadata pos "Function" bexp
 mkLambda (v:vs) bexp = do
-  pos <- asks currSrcPos
-  let funType = typeMdata pos "Function"
+  pos <- asks currentSrcPosition
+  let funType = typeMetadata pos "Function"
   funType . ELam v <$> mkLambda vs bexp
 
-isIf :: SimpleHL -> Maybe HnBodHC
+-- | A 'SimpleHL' represents an 'If' expression, if it is a single clause
+-- with a body.
+isIf :: SimpleHL -> Maybe HornBodyHeadClause
```

Comment:
I had been responsible for the `is<...>` naming convention for this sort of function (after being a bit too influenced by an episode of Unfoldr), and in retrospect I think that it was a mistake to have used `is<...>` for a function that does not return a boolean. ([This is what the Mirdin folks would call a name/type mismatch.](https://www.linguistic-antipatterns.com/?tab=%22Nametype-mismatch%22)) That is, if we want the return type to be a `Maybe`, then the name should probably be more like `getIf` rather than `isIf`.

I can make a PR that implements the `is<...>` -> `get<...>` changes if you'd like that.

Reply:
My gut feeling is, this reasoning doesn't apply in functional programming as much as in procedural programming. E.g. for `obj.isIf()` I would expect `boolean` as feel.

I don't feel like `isIf` is confusing, as it is very much like a `Bool`... Just that I have to run `if isIf exp then getIf exp else ...`, but it gives me a proof that it is indeed a `if` expression. `getIf` would imply to me that the operation can't really fail and doesn't perform arbitrarily complex work (but that's just my gut feeling). So `getIf :: ... Maybe a` feels less clear to me.
Some name alternatives I could get behind are `tryIf(Expr)` or `maybeIf(Expr)` (or `asIf(Expr)` but that feels a bit ambiguous), but I am personally perfectly happy with `isIf` in these scenarios.

Reply:
OK if you are happy with `is...` functions not returning a boolean, we can leave it as it is. I was just thinking of changing it because a senior engineer I was working with on a side project felt really strongly that functions named `is...` should return a boolean (the project didn't involve Haskell, but the senior engineer had a lot of experience in Haskell and PL as well).

Reply:
I am happy with it for now :)

Reply:
I am fine with `is...` functions for now :)